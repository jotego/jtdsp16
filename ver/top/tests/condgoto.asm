#Conditional goto
r0=0xf00
r1=0xf00
y=10
x=1
a0=y
a1=y
a1=a0-y    # EQ flag
if eq goto check1
r0=0xdead

check1:
a1=a0+y    # NEQ flag
if ne goto check2
r1=0xdead

check2:
# now check false conditions
r2=0xbad
a0=p
a0=a0+y
bad_end:
if eq goto bad_end

loop:
r2=0xcafe
goto loop

