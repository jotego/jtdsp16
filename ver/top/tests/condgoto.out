r0=0x0f00
r1=0x0f00
r2=0xcafe
r3=0x0000
rb=0x0000
re=0x0000
j =0x0000
k =0x0000
pr=0x0000
pt=0x0000
i =0x0000
a0=0x0000a0000
a1=0x000140000
x=0x0001
y=0x000a0000
p=0x00000000
c0=0x0000
c1=0x0000
c2=0x0000
auc=0x0000
psw=0x0000
